scripts/mixins/fomor_party.lua

```diff
@@ -0,0 +1,282 @@
+            { leader = phomiunaID.mob.FOMOR_THIEF[4],      followers = 1 },
+            { leader = phomiunaID.mob.FOMOR_WARRIOR[5],    followers = 1 },
+            { leader = phomiunaID.mob.FOMOR_RANGER[8],     followers = 1 },
+            -- Upstairs NW Room
+            { leader = phomiunaID.mob.FOMOR_BARD[6],       followers = 1 },
+            { leader = phomiunaID.mob.FOMOR_THIEF[6],      followers = 1 },
+            { leader = phomiunaID.mob.FOMOR_WARRIOR[3],    followers = 1 },
+            { leader = phomiunaID.mob.FOMOR_SAMURAI[4],    followers = 1 },
+            -- Downstairs L-7 Party
+            { leader = phomiunaID.mob.FOMOR_WARRIOR[1],    followers = 2 },
+            -- Downstairs I-7 Party
+            { leader = phomiunaID.mob.FOMOR_DRAGOON[1],    followers = 2 },
+        },
+
+        guard =
+        {
+            -- Tres Duendes Room
+            { members = { phomiunaID.mob.FOMOR_SAMURAI[1], phomiunaID.mob.FOMOR_NINJA[2], phomiunaID.mob.FOMOR_RANGER[1], phomiunaID.mob.FOMOR_DARK_KNIGHT[1] } },
+            -- H-8 Ladder
+            { members = { phomiunaID.mob.FOMOR_PALADIN[2], phomiunaID.mob.FOMOR_SAMURAI[2], phomiunaID.mob.FOMOR_RED_MAGE[1] } },
+            -- C-8 Ladder
+            { members = { phomiunaID.mob.FOMOR_BLACK_MAGE[2], phomiunaID.mob.FOMOR_WARRIOR[2], phomiunaID.mob.FOMOR_DARK_KNIGHT[3] } },
+        },
+    },
+}
+
+-----------------------------------
+-- Party Definition Helpers
+-----------------------------------
+local function getZoneParties(mob)
+    local zoneID = mob:getZoneID()
+    return zoneParties[zoneID]
+end
+
+local function getFomorParty(mobID, parties)
+    if not parties then
+        return nil, nil
+    end
+
+    if parties.patrol then
+        for _, party in ipairs(parties.patrol) do
+            if
+                party.leader and
+                party.followers and
+                mobID >= party.leader and
+                mobID <= party.leader + party.followers
+            then
+                return party, 'patrol'
+            end
+        end
+    end
+
+    if parties.guard then
+        for _, party in ipairs(parties.guard) do
+            if party.members then
+                for _, memberID in ipairs(party.members) do
+                    if memberID == mobID then
+                        return party, 'guard'
+                    end
+                end
+            end
+        end
+    end
+
+    return nil, nil
+end
+
+local function getPartyLeader(party)
+    if party then
+        if party.leader then
+            return party.leader
+        elseif party.members and #party.members > 0 then
+            return party.members[1]
+        end
+    end
+
+    return nil
+end
+
+-----------------------------------
+-- Set up party behavior on spawn
+-- Handles superlinking, guard/patrol behavior, and following
+-----------------------------------
+xi.mix.fomorParty.onPartySpawn = function(mob, parties)
```

Comment:
There is not a single case-use of the second argument (parties) of this functions.
`xi.mix.fomorParty.onPartySpawn()`
`xi.mix.fomorParty.onPartyRoam()`
`xi.mix.fomorParty.onPartyDeath()`

So remove them from here.

Reply:
Removed the unnecessary getZoneParties function and removed the parties argument from all the functions listed.